src/EditorFeatures/Core/Diagnostics/AbstractPushOrPullDiagnosticsTaggerProvider.SingleDiagnosticKindPullTaggerProvider.cs

```diff
@@ -155,6 +158,25 @@ protected sealed override ITaggerEventSource CreateEventSource(ITextView? textVi
             }
         }
 
+        private bool HasSuccessfullyLoaded(Project? project, HashSet<Project> seenProjects)
+        {
+            if (project != null && seenProjects.Add(project))
+            {
+                if (!project.State.HasAllInformation)
+                    return false;
+
+                // Ensure our dependencies have all information as well.  That's necessary so we can properly get
+                // compilations for them.
+                foreach (var reference in project.ProjectReferences)
```

Comment:
What about circular project references for error cases? Will this code go into infinite recursion?

Reply:
That's what the set is for. To prevent recursion.